File: fixtures/scale/java/large/src/cryptobench/ciphers/route/RouteSuite.java
package cryptobench.ciphers.route;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class RouteSuite implements CipherSuite {
    @Override
    public String name() {
        return "route";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new RouteCipher(RouteKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : RouteVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/route/RouteSuite.kt
package cryptobench.ciphers.route

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class RouteSuite : CipherSuite {
    override fun name(): String = "route"

    override fun run(): SuiteResult {
        val cipher: Cipher = RouteCipher(RouteKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in RouteVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
